.github/workflows/main.yml

```diff
@@ -31,9 +31,23 @@ jobs:
           path: ~/.m2/repository
           key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
-            ${{ runner.os }}-maven-
+            ${{ runner.os }}-maven-      
       - name: Build with Maven
-        run: mvn verify --batch-mode -P integration-test --file pom.xml
+        run: |
+          mvn verify --batch-mode -P integration-test --file pom.xml
+      - name: run db and web containers (ubuntu only)
+        if: matrix.platform == 'ubuntu-latest'
+        run: |
+          docker-compose -f docker/docker-compose-refqa.yml up -d
+      - name: wait for openmrs instance to start (ubuntu only)
+        if: matrix.platform == 'ubuntu-latest'
+        run: while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' http://localhost:8080/openmrs/login.htm)" != "200" ]]; do sleep 1; done
+      - name: Run End to End tests (ubuntu only)
+        if: matrix.platform == 'ubuntu-latest'
+        run: mvn -pl e2e-tests -amd verify --batch-mode -P e2e-test --file pom.xml
+      - name: Stop db and web containers (ubuntu only)
+        if: always() && matrix.platform == 'ubuntu-latest'
+        run: docker-compose -f "docker/docker-compose-refqa.yml" down
```

Comment:
Awesome! But maybe we should separate this out into another job? That's got a few advantages:

1. We can use a single `runs-on` instead of a bunch of tests
2. We can separately re-run the job instead having to re-run everything

We'd likely need to setup outputs for the `build` job so we can use them in the other job (see [GitHub's docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idoutputs=))

Reply:
> Awesome! But maybe we should separate this out into another job?

Using a different job makes things prettier

Reply:
hey @ibacher 
it seamed [github's outputs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idoutputs) only stored strings yet i needed to share the results of an omod build so i uploaded the results of  the omod build and then downloaded it in another workflow

I am sure this approach has some flows but i wish to know your suggestions at this point 
cc @dkayiwa